src/metabase/query_processor/middleware/large_int_id.clj

```diff
@@ -0,0 +1,47 @@
+(ns metabase.query-processor.middleware.large-int-id
+  "Middleware for handling conversion of IDs to strings for proper display of large numbers"
+  (:require [metabase.mbql.util :as mbql.u]
+            [metabase.query-processor.store :as qp.store]))
+
+(defn- result-int->string
+  [field-indexes rf]
+  (fn
+    ([] (rf))
+    ([result] (rf result))
+    ([result row]
+     (if (and (seq field-indexes)
+              row)
+       (rf result (reduce #(update %1 %2 str) row field-indexes))
+       (rf result row)))))
+
+(defn convert-id-to-string
+  "Converts any ID (:type/PK and :type/FK) in a result to a string to handle a number > 2^51
+  or < -2^51, the JavaScript float mantissa. This will allow proper display of large numbers,
+  like IDs from services like social media. All ID numbers are converted to avoid the performance
+  penalty of a comparison based on size."
+  [qp]
+  (fn [{{:keys [js-int-to-string?] :or {js-int-to-string? false}} :middleware, :as query} rff context]
+    ;; currently, this excludes `:field-literal` values like aggregations.
```

Comment:
Not converting aggregated columns seem fine to me. If you sum some IDs, you probably don't have an ID anymore.

I think native queries could be an issue though. I guess fixing it for structured queries is better than not having a solution at all, but this might push us to the "full" solution more quickly.

Reply:
Yeah, native queries could be a problem. On a native query, we don't know what's an ID, we just know it's an Integer type.